server/src/main/java/org/opensearch/search/aggregations/AggregatorFactories.java

```diff
@@ -257,6 +257,10 @@ private AggregatorFactories(AggregatorFactory[] factories) {
         this.factories = factories;
     }
 
+    public AggregatorFactory[] getFactories() {
+        return factories;
+    }
+
```

Comment:
instead of returning the `factories`, we can add a method `allFactoriesSupportConcurrentSearch` here as well which can iterate over all the `AggregationFactory`

```
public boolean allFactoriesSupportConcurrentSearch() {
for (AggregationFactory factory : factories) {
    if (!factory.supportsConcurrentSegmentSearch()) {
         return false;
    }
}
return true; 
}
```